ldi-core/ldes-client/tree-node-supplier/src/main/java/ldes/client/treenodesupplier/repository/filebased/FileBasedMemberRepository.java

```diff
@@ -0,0 +1,57 @@
+package ldes.client.treenodesupplier.repository.filebased;
+
+import ldes.client.treenodesupplier.domain.entities.MemberRecord;
+import ldes.client.treenodesupplier.domain.valueobject.MemberStatus;
+import ldes.client.treenodesupplier.repository.MemberRepository;
+import ldes.client.treenodesupplier.repository.filebased.mapper.MemberRecordMapper;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class FileBasedMemberRepository implements MemberRepository {
+	public static final String UNPROCESSED_MEMBERS = "membersUnprocessed.txt";
+	public static final String PROCESSED_MEMBERS = "membersProcessed.txt";
+
+	private final FileManagerFactory fileManagerFactory = FileManagerFactory.getInstance();
+	private final FileManager fileManager = fileManagerFactory.getFileManager();
+	private final MemberRecordMapper mapper = new MemberRecordMapper();
+
+	@Override
+	public Optional<MemberRecord> getUnprocessedTreeMember() {
+		List<MemberRecord> unprocessed = fileManager.getRecords(UNPROCESSED_MEMBERS).map(mapper::toMemberRecord)
+				.collect(Collectors.toList());
+		if (unprocessed.isEmpty()) {
+			return Optional.empty();
+		}
+		return Optional.of(unprocessed.remove(0));
+	}
+
+	@Override
+	public boolean isProcessed(MemberRecord member) {
+		List<MemberRecord> processed = fileManager.getRecords(PROCESSED_MEMBERS).map(mapper::toMemberRecord).toList();
```

Comment:
How big can this file get? Parsing models is usually the biggest performance hit in our code. This method does not require any of those models. Maybe a 'getRecordIds' could help here? Just checking the file if it contains the id may be to error prone if some model has a pointer to the member with that id.

If these are big files, we may want to use a stream instead of a list.

Reply:
Once, a member is processed its model is set to `null` so the file memberProcessed.txt only contains ids en statusses

Reply:
But even then it could indeed get pretty bit, deleted the collector :-)

Reply:
I did a quick test, we have a 50 MB file after 500k members. At this point it takes 2-3 minutes to process 150 members on my laptop with 32GB memory and i7 cpu